File: Client/Captions/CaptionsLib/source/models/ICaptionsReader.vb
'Filename: ICaptionsReader.vb
'Version: 20121016

Imports System.IO
Imports System.Text
Imports Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions

Namespace ClipFlair.CaptionsLib.Models

  Public Interface ICaptionsReader

    'not clearing any existing captions, just adding new ones (the CaptionRegion object can choose whether it will sort the Captions by start time or not after the appending)
    Sub ReadCaptions(ByVal captions As CaptionRegion, ByVal path As String, ByVal theEncoding As Encoding)
    Sub ReadCaptions(ByVal captions As CaptionRegion, ByVal stream As Stream, ByVal theEncoding As Encoding)
    Sub ReadCaptions(ByVal captions As CaptionRegion, ByVal reader As TextReader)

    Sub ReadHeader(ByVal reader As TextReader)
    Sub ReadCaption(ByVal caption As CaptionElement, ByVal reader As TextReader)
    Sub ReadFooter(ByVal reader As TextReader)

  End Interface

End Namespace

File: Client/Captions/CaptionsLib/source/models/ICaptionsReader.cs
//Project: ClipFlair (http://ClipFlair.codeplex.com)
//Filename: ICaptionsReader.cs
//Version: 20131105

using System.IO;
using System.Text;
using Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions;

namespace ClipFlair.CaptionsLib.Models
{

	public interface ICaptionsReader
	{

		//not clearing any existing captions, just adding new ones (the CaptionRegion object can choose whether it will sort the Captions by start time or not after the appending)
		void ReadCaptions<T>(CaptionRegion captions, string path, Encoding theEncoding) where T : CaptionElement, new();
		void ReadCaptions<T>(CaptionRegion captions, Stream stream, Encoding theEncoding) where T : CaptionElement, new();

		void ReadCaptions<T>(CaptionRegion captions, TextReader reader) where T : CaptionElement, new();
		void ReadHeader(TextReader reader);
		void ReadCaption(CaptionElement caption, TextReader reader);

		void ReadFooter(TextReader reader);
	}

}